src/main/scala/exercises/07_Palindrome.scala

```diff
@@ -8,6 +8,10 @@ object Palindrome {
    *  Hint: the Scala collections API is your friend
    *  http://docs.scala-lang.org/overviews/collections/overview.html 
    */
-  def isPalindrome(s: String): Boolean = throw new NotImplementedError
+  def isPalindrome(s: String): Boolean = {
+    // remove punctuation and convert every letter to lowercase
+    val data = s.filter(_.isLetter).map(_.toLower)
+    data == data.reverse
+  }
```

Comment:
Wow this is much cleaner than what I did.  I remember writing this function in CS70, where we first removed punctuation and converted everything to lower, as you did.  But then we passed the result to a helper function that checked to see if the first and last letters were the same, stripped them away, and recursed.  This is much more concise.